Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -108,14 +110,17 @@ ALWAYS_INLINE void ContainerNode::removeAllChildrenWithScriptAssertion(ChildChan
         ScriptDisallowedScope::InMainThread scriptDisallowedScope;
         if (UNLIKELY(document().hasMutationObserversOfType(MutationObserverOptionType::ChildList))) {
             ChildListMutationScope mutation(*this);
-            for (auto& child : children)
+            for (auto& child : children) {
+                hadElementChild |= is<Element>(child);
                 mutation.willRemoveChild(child.get());
-        }
+            }
+        } else
+            hadElementChild = children.containsIf([](auto& child) { return is<Element>(child); });
     }
 
     disconnectSubframesIfNeeded(*this, DescendantsOnly);
 
-    ContainerNode::ChildChange childChange { ChildChange::Type::AllChildrenRemoved, nullptr, nullptr, nullptr, source };
+    ContainerNode::ChildChange childChange { ChildChange::Type::AllChildrenRemoved, nullptr, nullptr, nullptr, source, hadElementChild ? ContainerNode::ChildChange::AffectsElements::Yes : ContainerNode::ChildChange::AffectsElements::No };
```

Comment:
Why don't we make this function return a value instead of iterating over removed children once again in ContainerNode::replaceAll?

Reply:
Great idea, will do.